File: OBI/2021/M2/lista.c
#include<stdio.h>
//
int main(){
    int n;
    scanf("%i",&n);

    int arr[n];
    for(size_t i=0;i<(size_t)n;++i)
        scanf("%i",&arr[i]);

    size_t left=0,right=n-1;
    int left_value=arr[left],
        right_value=arr[right],
        moves=0;
    while(1){
        if(left==right)
            break;
        if(left_value==right_value&&right-left==1)
            break;
        if(left_value==right_value){
            ++left;
            --right;

            left_value=arr[left];
            right_value=arr[right];
            continue;
        }

        if(left_value<right_value){
            ++left;
            ++moves;
            left_value+=arr[left];
            continue;
        }
        if(left_value>right_value){
            --right;
            ++moves;
            right_value+=arr[right];
            continue;
        }
    }

    printf("%i\n",moves);

    return 0;
}

File: OBI/2021/M2/Level_2/lista.py
def main()->None:
    x=input()
    array=list(map(int,input().split()))

    left,right=0,len(array)-1
    left_value,right_value=array[left],array[right]
    moves=0
    while(1):
        if left==right:
            break
        if right-left==1 and right_value==left_value:
            break
        if right_value==left_value:
            left+=1
            right-=1
            left_value=array[left]
            right_value=array[right]
            continue

        if left_value<right_value:
            left+=1
            moves+=1
            left_value+=array[left]
            continue
        if left_value>right_value:
            right-=1
            moves+=1
            right_value+=array[right]
            continue

    print(moves)

print(__name__)
if __name__=='__main__':
    main()
